docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+  parameters and output formats
+
+This approach was selected as the optimal balance between security,
+functionality, and development effort.
+
+## Security Model and RBAC Constraints
+
+No matter how we approach this implementation, it is impossible to totally
+secure the execution of networking troubleshooting tools at this stage. Most
+networking tools require privileged access to function properly, creating
+inherent security trade-offs. The goal is therefore to minimize blast radius
+through layered controls rather than achieve perfect isolation.
+
+**Kubernetes Layer Security:**
+
+* `kubectl exec` and `oc debug node` operations require cluster-admin level
+  privileges by design
+* Alternative approach using custom debug containers
+  (`oc debug node --image=&lt;custom-image>`) with volume mounts to database
+  files reduces some attack surface but remains intrusive
+* **Mitigation**: Strict command allowlisting - expose only read operations
+  (`get`, `list`, `describe`, `logs`) while blocking any modification commands
+* **Constraint**: MCP Server service account requires elevated privileges
+  (including `pods/exec`) despite being conceptually a "troubleshooter" role
+
+For the kubernetes layer, we are going to leverage the opensource
+[kubernetes-mcp-server](https://github.com/containers/kubernetes-mcp-server).
+So whatever security posture they use for now, can be adopted.
+
+**OVN/OVS Database Layer Security:**
+
+* Unix socket-based database access prevents using SSL certificate-based
+  authentication and authorization
+* Database connections inherit the security context of the container executing
+  the commands
+* **Mitigation**: Command parameter validation to ensure only read-only
+  database operations (`show`, `list`, `dump`) while blocking modification
+  commands (`set`, `add`, `remove`)
+* **Long-term Path**: Requires RBAC-enabled CLI execution even against local
+  unix socket.
+
+**Host/Kernel Layer Security:**
+
+* Kernel-level networking tools (`ip`, `nft`, `conntrack`) inherently require
+  root system access
+* Current tooling lacks granular RBAC capabilities - tools are typically
+  all-or-nothing from a privilege perspective
+* **External Tools Note**: Tools like `tcpdump -i any` can be highly intrusive
+  as they capture all network traffic on the host, requiring careful
+  consideration of privacy and performance impact while being chosen for
+  execution.
+* **Short-term Mitigation**: Strict command allowlisting exposing only read
+  operations (`ip route show`, `nft list ruleset`) while blocking modification
+  commands (`ip route add`, `nft add rule`)
+* **Long-term Path**: Requires RBAC-enabled wrapper tools from upstream
+  layered product teams (example netfilter, kernel networking)
+
+## Distributed Execution Context
+
+### **OVN-Kubernetes Architecture Challenge**
+
+In OVN-Kubernetes interconnect architecture, each node maintains its own local
```

Comment:
I assume we are not considering the MCP for the centralized architecture right?

Reply:
yea nope. I'll call that out.